pkg/registry/flowcontrol/flowschema/strategy.go

```diff
@@ -0,0 +1,102 @@
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol/validation"
+)
+
+// flowSchemaStrategy implements verification logic for FlowSchema.
+type flowSchemaStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating flow schema objects.
+var Strategy = flowSchemaStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// NamespaceScoped returns false because all PriorityClasses are global.
+func (flowSchemaStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate clears the status of a flow-schema before creation.
+func (flowSchemaStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	fl := obj.(*flowcontrol.FlowSchema)
+	fl.Status = flowcontrol.FlowSchemaStatus{}
+	fl.Generation = 1
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (flowSchemaStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newFlowSchema := obj.(*flowcontrol.FlowSchema)
+	oldFlowSchema := old.(*flowcontrol.FlowSchema)
+
+	// Spec updates bump the generation so that we can distinguish between status updates.
+	if !apiequality.Semantic.DeepEqual(newFlowSchema.Spec, oldFlowSchema.Spec) {
+		newFlowSchema.Generation = oldFlowSchema.Generation + 1
+	}
+	newFlowSchema.Status = oldFlowSchema.Status
+}
+
+// Validate validates a new flow-schema.
+func (flowSchemaStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
+	return validation.ValidateFlowSchema(obj.(*flowcontrol.FlowSchema))
+}
+
+// Canonicalize normalizes the object after validation.
+func (flowSchemaStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (flowSchemaStrategy) AllowUnconditionalUpdate() bool {
+	return true
+}
+
+// AllowCreateOnUpdate is false for flow-schemas; this means a POST is needed to create one.
+func (flowSchemaStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// ValidateUpdate is the default update validation for an end user.
+func (flowSchemaStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return validation.ValidateFlowSchema(obj.(*flowcontrol.FlowSchema))
+}
+
+type flowSchemaStatusStrategy struct {
+	flowSchemaStrategy
+}
+
+// StatusStrategy is the default logic that applies when updating flow-schema objects' status.
+var StatusStrategy = flowSchemaStatusStrategy{Strategy}
+
+func (flowSchemaStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newFlowSchema := obj.(*flowcontrol.FlowSchema)
+	oldFlowSchema := old.(*flowcontrol.FlowSchema)
+	newFlowSchema.Spec = oldFlowSchema.Spec
+}
+
+func (flowSchemaStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return validation.ValidateFlowSchema(obj.(*flowcontrol.FlowSchema))
```

Comment:
You should `ValidateFlowSchemaStatusUpdate` in particular.  Create the function even if it has no special values today.  Recall that you should not validate spec while updating status or vice-versa because changing rules on unrelated values can break old clients.

Reply:
done. following the pattern of pod status update strategy.